### ISSUE TITLE
✨ Enhancement: Accessibility Enhancements for All Components

### ISSUE THREAD

**Issue by shivam8112005** [json-schema-org/website#1195]
### Is your feature request related to a problem? Please describe

Components lack proper accessibility features like ARIA attributes and semantic roles, making them less usable for assistive technology users.

### Describe the solution you'd like

Add ARIA attributes, semantic roles, and keyboard navigation support to all components, including Accordion, Buttons, Forms, Modals, Navigation, and Images.

### Describe alternatives you've considered

Using libraries or fixing issues one component at a time.

### Additional context

Making components accessible benefits all users.

### Are you working on this?

Yes

**Comment by heysujal** [comment]
This sounds great.

**Comment by shivam8112005** [comment]
Yepp

**Comment by shivam8112005** [comment]
Assign this to me

**Comment by heysujal** [comment]
I am not a maintainer for this project. I think you should create another seperate issue(s) for a specific component and then ask to claim it.

Tagging @DhairyaMajmudar

**Comment by benjagm** [comment]
Good proposal. Can ytou provide an example of what implementation of what your are proposing here?

**Comment by shivam8112005** [comment]
@benjagm Thanks

![Screenshot 2024-12-23 155555](https://github.com/user-attachments/assets/b5963ce3-0828-421e-9163-3aed142a77c4)


This is just an example in the Accordion component where I added:

role="tablist" on the container for defining the tab group.
role="tab" and aria-expanded on each question for accessibility.
aria-controls on each question, linking it to its respective answer panel.
role="tabpanel" on each answer panel.
aria-labelledby on each panel, linking it back to the question.
aria-label on the links for providing additional context.

**Comment by shivam8112005** [comment]
Hi @benjagm  just following up on this issue  #1195  —let me know if any further details are needed from my side. Thanks!

**Comment by benjagm** [comment]
We are clossing this issues as it will be worked out by @idanidan29 as part of GSoC.

**Comment by shivam8112005** [comment]
@benjagm Ok No Problem
